ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -190,34 +198,46 @@ A single compaction job could be performed across multiple compaction cycles (if
     4. CLEANUP - Cleans up the old log segments and their index files
      */
 
+    logger.trace("Resuming compaction");
     try {
+      compactionInProgress.set(true);
```

Comment:
this can be set before entering the `try` block? Just more readable if the logging and the flag setting are together.